### PR TITLE
Bi-5224 allow to disable joins optimization

### PR DESCRIPTION
**Pull request by AdayevKP**

### CODE REVIEW

File: lib/dl_api_connector/dl_api_connector/api_schema/source_base.py [hunk 1]
```diff
@@ -288,3 +288,4 @@ def make_condition(self, data: dict, **kwargs: Any) -> BinaryCondition:
     join_type = ma_fields.Enum(JoinType)
     managed_by = ma_fields.Enum(ManagedBy, allow_none=True, dump_default=ManagedBy.user, load_default=ManagedBy.user)
     virtual = VirtualFlagField(attribute="managed_by", dump_only=True)
+    required = ma_fields.Bool(load_default=False, required=False, dump_default=True)
```
Comment (ovsds) [inline lib/dl_api_connector/dl_api_connector/api_schema/source_base.py:291]: why different load and dump defaults?

File: lib/dl_core_testing/dl_core_testing/dataset_builder.py [hunk 2]
```diff
@@ -189,29 +186,23 @@ def add_source_avatar(self, source_id: str) -> SourceAvatarProxy:
...
             relation_id=relation_id,
         )
 
     def add_avatar_relation_simple_eq(
-        self,
-        left_avatar_id: str,
-        right_avatar_id: str,
-        left_col_name: str,
-        right_col_name: str,
+        self, left_avatar_id: str, right_avatar_id: str, left_col_name: str, right_col_name: str, required: bool = False
```
Comment (ovsds) [inline lib/dl_core_testing/dl_core_testing/dataset_builder.py:205]: add comma in the end so it would stay multilined

Comment (AdayevKP) [reply lib/dl_core_testing/dl_core_testing/dataset_builder.py]: done

File: lib/dl_api_connector/dl_api_connector/api_schema/source_base.py (hunk 1 above)
Comment (AdayevKP) [reply lib/dl_api_connector/dl_api_connector/api_schema/source_base.py]: it's a mistake. Fixed, now both default is `False`

**Review by ovsds — APPROVED** [review]

**Comment by AdayevKP** [comment]
tests failed not due to PR changes

**Pull request merged**